lib/matplotlib/__init__.py

```diff
@@ -496,7 +496,12 @@ def matplotlib_fname():
     """
 
     def gen_candidates():
-        yield os.path.join(os.getcwd(), 'matplotlibrc')
+        try:
+            # if the user has ended up with a non-existent directory
+            # as the current working directory do not fail to import.
+            yield os.path.join(os.getcwd(), 'matplotlibrc')
+        except FileNotFoundError:
+            pass
```

Comment:
```suggestion
        # rely on down-stream code to make absolute.  This protects us
        # from having to directly get if the current working directory
        # which can fail if the user has ended up with a cwd that is
        # non-existent.
        yield 'matplotlibrc'
```

Also works.  It is a bit simpler (no try-except) but marginally higher risk (as we are changing the cases when this is the expected behavior).

Reply:
I like this version.